src/vs/workbench/parts/search/browser/searchActions.ts

```diff
@@ -317,7 +317,7 @@ export class ClearSearchResultsAction extends Action {
 
 	update(): void {
 		const searchView = getSearchView(this.viewletService, this.panelService);
-		this.enabled = searchView && searchView.isSearchSubmitted();
+		this.enabled = searchView && !searchView.allSearchFieldsClear();//searchView.isSearchSubmitted();
```

Comment:
You should leave the check for isSearchSubmitted. Suppose the user made a search, than cleared the inputs (manually). Now he should be able to click the search action in order to clean the result view.

Reply:
You should check if the results view is clear.

Reply:
Corrected in latest commit.